subsys/settings/src/settings_static_data.c

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright (c) 2024 Laczen
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <errno.h>
+#include <string.h>
+#include <zephyr/settings/settings.h>
+
+#include <zephyr/logging/log.h>
+LOG_MODULE_DECLARE(settings_static_data, CONFIG_SETTINGS_LOG_LEVEL);
+
+static ssize_t settings_static_data_read_fn(void *back_end, void *data, size_t len)
+{
+	struct settings_static_data *static_data =
+		(struct settings_static_data *)back_end;
+	ssize_t rc = (len > static_data->size) ? -EINVAL : len;
+
+	if (rc > 0) {
+		memcpy(data, static_data->data, len);
```

Comment:
What if we had a "zero-copy" option of the read callback to further reduce the overall memory and CPU overhead of using the settings subsystem vs. using the macro target. This callback would only be made available by backends that actually have this capability so that the client (e.g. the configured subsystem) knows how to handle memory ownership on its side.

Reply:
The test is only showing the basic idea. The exchanged data could (in a configuration solution) be limited to pointers to the config data. The copy operation would then only be a copy of the pointers, not of the data.